fastapi/encoders.py

```diff
@@ -240,7 +240,7 @@ def jsonable_encoder(
             custom_encoder=encoders,
             sqlalchemy_safe=sqlalchemy_safe,
         )
-    if dataclasses.is_dataclass(obj):
+    if dataclasses.is_dataclass(type(obj)):
```

Comment:
Is this needed? :thinking:

Reply:
Yep, see the commit message.

Reply:
https://github.com/python/typeshed/blob/e1c74f08f12b8bec6afe266951f0756dc1b43ebe/stdlib/dataclasses.pyi#L217-L223

Reply:
@tiangolo how can I move this forward?